Source/ASDisplayNode.mm

```diff
@@ -1220,11 +1220,10 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
   }
   ASDisplayNodeLogEvent(self, @"computedLayout: %@", layout);
 
-  // Return the (original) unflattened layout if it needs to be stored. The layout will be flattened later on (@see _locked_setCalculatedDisplayNodeLayout:).
-  // Otherwise, flatten it right away.
-  if (! [ASDisplayNode shouldStoreUnflattenedLayouts]) {
-    layout = [layout filteredNodeLayoutTree];
+  if ([ASDisplayNode shouldStoreUnflattenedLayouts]) {
+      _unflattenedLayout = layout;
```

Comment:
As discussed offline, there is a small risk that the layout here will not be used/applied later on, and Weaver (or any other users of this debugging feature) will end up with a wrong unflattened layout. But given it's a debugging feature and should not be used in producation, the risk is small and doesn't warrant the complexity in previous implementation.

Reply:
Can you add some comment about this fact?